api/service/stagedstreamsync/staged_stream_sync.go

```diff
@@ -583,3 +594,133 @@ func (s *StagedStreamSync) EnableStages(ids ...SyncStageID) {
 		}
 	}
 }
+
+func (ss *StagedStreamSync) purgeLastMileBlocksFromCache() {
+	ss.lastMileMux.Lock()
+	ss.lastMileBlocks = nil
+	ss.lastMileMux.Unlock()
+}
+
+// AddLastMileBlock adds the latest a few block into queue for syncing
+// only keep the latest blocks with size capped by LastMileBlocksSize
+func (ss *StagedStreamSync) AddLastMileBlock(block *types.Block) {
+	ss.lastMileMux.Lock()
+	defer ss.lastMileMux.Unlock()
+	if ss.lastMileBlocks != nil {
+		if len(ss.lastMileBlocks) >= LastMileBlocksSize {
+			ss.lastMileBlocks = ss.lastMileBlocks[1:]
+		}
+		ss.lastMileBlocks = append(ss.lastMileBlocks, block)
+	}
+}
+
+func (ss *StagedStreamSync) getBlockFromLastMileBlocksByParentHash(parentHash common.Hash) *types.Block {
+	ss.lastMileMux.Lock()
+	defer ss.lastMileMux.Unlock()
+	for _, block := range ss.lastMileBlocks {
+		ph := block.ParentHash()
+		if bytes.Equal(ph[:], parentHash[:]) {
```

Comment:
nit: you don't need bytes.Equal for arrays of equal length. Just `==` works fine and is marginally faster.

Reply:
done